blobfuse/fileapis.cpp

```diff
@@ -84,10 +84,10 @@ int azs_open(const char *path, struct fuse_file_info *fi)
     auto fmutex = file_lock_map::get_instance()->get_mutex(path);
     std::lock_guard<std::mutex> lock(*fmutex);
 
-    // If the file/blob being opened does not exist in the cache, or the version in the cache is too old, we need to download / refresh the data from the service.
+    // If the file/blob being opened does not exist in the cache, or the version in the cache is too old, or it occupies zero blocks with a size (List attribute cache), we need to download / refresh the data from the service.
```

Comment:
are you sure that's true that ftruncated file doesn't occupy blocks on every filesystem?

Reply:
Ftruncate fills the with zeros: https://linux.die.net/man/2/ftruncate

Zeroes (file holes) are not supposed to take up space in UNIX systems. Though haven't seen any guarantees in the POSIX spec. Any ideas how we can validate this ? Or is there a better approach ?

Reply:
i believe it all depends on if file system support sparse files or not. At least some time about tmpfs didn't support sparse files, some ntfs drivers didn't, fat didn't, ext3,4 do. i remember there were filesystems which supported sparse files but still allocated one or two blocks.
if you would like to use this approach, you'll better give requirement for cache file system.

Reply:
Now have a sparse file test at the startup. If sparse files are not supported, then the list attribute cache gets disabled.